src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -86,38 +69,69 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
 
-  if (previousSelectedPath !== selectedPath) {
-    setIsNewSelectedPath(true);
-    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
-    setPreviousSelectedPath(selectedPath);
-  }
+  /**
+   * Assigns focus to current node.
+   */
+  const focusCurrentNode = () => {
+    const currentNode = currentNodeId.current ? document.getElementById(currentNodeId.current) : null;
+    if (currentNode) {
+      currentNode.focus();
+    }
+  };
 
+  /**
+   * Scrolls the currently selected menu item into view on mount.
+   * Ensures that the cursor is synched with the currently selected item.
+   */
   useEffect(() => {
+    const nodeId = selectedItem.current?.id;
+    const idx = visibleNodes.findIndex((el) => el.id === nodeId);
+    if (idx >= 0) {
+      cursor.current = idx;
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
     if (selectedItem && selectedItem.current) {
       selectedItem.current.scrollIntoView();
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  useLayoutEffect(() => {
-    if (!isNewSelectedPath) {
-      return;
-    }
-    const currentItemPosition = selectedItem?.current ? selectedItem.current.getBoundingClientRect() : null;
-    const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
-    if (!currentItemPosition || !navigationMenuPosition) {
-      return;
+  /**
+   * Ensures that the cursor is synched with the currently selected item.
+   * Scrolls selected item into view.
+   */
+  useEffect(() => {
+    if (currentNodeId.current) {
+      cursor.current = visibleNodes.findIndex((el) => el.id === currentNodeId.current);
     }
-    // If the current item is not visible, scroll the item into view.
-    if (currentItemPosition.bottom > navigationMenuPosition.bottom || currentItemPosition.top < navigationMenuPosition.top) {
-      selectedItem.current.scrollIntoView();
+
+    if (previousSelectedPath.current !== selectedPath) {
+      const selectedItemPosition = selectedItem?.current ? selectedItem.current.getBoundingClientRect() : null;
+      const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
+      if (selectedItemPosition && navigationMenuPosition && (selectedItemPosition.bottom > navigationMenuPosition.bottom || selectedItemPosition.top < navigationMenuPosition.top)) {
+        selectedItem.current.scrollIntoView();
+      }
+      previousSelectedPath.current = selectedPath;
     }
-  });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [openKeys]);
+
+  /**
+   * Ensures the parent of the currently selected item is expanded.
+   */
+  useEffect(() => {
+    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedPath]);
 
   const handleOnClick = (event, item) => {
```

Comment:
I noticed that clicking an item doesn't move the cursor position to that item. It seems like it should.

Reply:
I agree, I think this was in a previous version of this branch, but it got lost somewhere along the way. I've inadvertently added this functionality in addressing some of Dave's comments.